src/net/http.rs

```diff
@@ -1073,6 +1174,33 @@ impl HttpRequestType {
 
         Ok(HttpRequestType::GetMicroblocks(HttpRequestMetadata::from_preamble(preamble), block_hash))
     }
+    
+    fn parse_getmicroblocks_unconfirmed<R: Read>(_protocol: &mut StacksHttp, preamble: &HttpRequestPreamble, regex: &Regex, _fd: &mut R) -> Result<HttpRequestType, net_error> {
+        if preamble.get_content_length() != 0 {
+            return Err(net_error::DeserializeError("Invalid Http request: expected 0-length body for GetMicrolocksUnconfirmed".to_string()));
+        }
+
+        let block_hash_str = regex
+            .captures(&preamble.path)
+            .ok_or(net_error::DeserializeError("Failed to match path to microblock hash".to_string()))?
+            .get(1)
+            .ok_or(net_error::DeserializeError("Failed to match path to microblock hash group".to_string()))?
+            .as_str();
+
+        let min_seq_str = regex
+            .captures(&preamble.path)
```

Comment:
You should bind `regex.captures(&preamble.path)` to a variable so that it only needs to run once.

Reply:
Thanks; will do :+1:

Reply:
Fixed.